### PR TITLE
deps: Update syntex_syntax to 0.31.0

### PR DESCRIPTION
**Pull request by kamalmarhubi**
Most of the churn on this bump comes from the `Visibility` enum changing
from

    pub enum Visibility {
        Public,
        Inherited,
    }

to

    pub enum Visibility {
        Public,
        Crate,
        Restricted { path: P<Path>, id: NodeId },
        Inherited,
    }

which require taking `Visibility` by reference in most places. The new
variants are not handled at this point.

Refs #970

### CODE REVIEW

File: src/utils.rs [hunk 1]
```diff
@@ -67,10 +67,13 @@ pub fn extra_offset(text: &str, offset: Indent) -> usize {
...
 #[inline]
-pub fn format_visibility(vis: Visibility) -> &'static str {
-    match vis {
+pub fn format_visibility(vis: &Visibility) -> &'static str {
+    match *vis {
         Visibility::Public => "pub ",
         Visibility::Inherited => "",
+        // TODO(#970): Handle new visibility types.
+        Visibility::Crate => unimplemented!(),
+        Visibility::Restricted { .. } => unimplemented!(),
```
Comment (nrc) [inline src/utils.rs]: Could we handle these by returning None and then skipping the item, falling back to the missing spans? That is better than panicking, I think.

Also, (nit) if there is an issue number, prefer `FIXME` to `TODO`.

**Comment by nrc** [comment]
All looks good, other than the comment on the panics, thanks!

File: src/utils.rs (hunk 1 above)
Comment (kamalmarhubi) [inline src/utils.rs]: Done!

**Comment by marcusklaas** [comment]
We can land this, but we need syntex 0.32 for the try shorthand. [This heroic pull request by antrik](https://github.com/serde-rs/syntex/pull/44) includes the required changes.

**Comment by kamalmarhubi** [comment]
I'm happy to do another bump. I kind of enjoy the compiler-error chasing that goes with these bumps. Nice break from other stuff sometimes!

**Comment by kamalmarhubi** [comment]
Assuming this is acceptable as an approach for handling the `Visibility` change, it'd be great to merge this and then see what 0.32 brings.

**Pull request merged**